### PR TITLE
Quote rsync transport safely

### PR DESCRIPTION
**Pull request by salva**
## Summary

Make rsync transport command quoting safe for POSIX shell command strings.

## Changes

- Quote every transport argument with POSIX single-quote escaping.
- Preserve embedded single quotes.
- Double all `%` characters, not just the first one.
- Add focused regression tests for whitespace, single quotes, percent escaping, and list context.
- Add the new test to `MANIFEST`.

Fixes #41.

## Testing

- `perl -Ilib -c lib/Net/OpenSSH.pm`
- `perl -Ilib t/rsync-quote.t`

### CODE REVIEW

File: lib/Net/OpenSSH.pm [hunk 1]
```diff
@@ -640,11 +640,9 @@ sub _make_scp_call {
...
     my ($self, @args) = @_;
     for (@args) {
-	if (/['"\s]/) {
-	    s/"/""/g;
-	    $_ = qq|"$_"|;
-	}
-	s/%/%%/;
+        s/%/%%/g;
+        s/'/'\\''/g;
+        $_ = qq|'$_'|;
```
[the remark this answers is not in the document]
Comment (salva) [reply lib/Net/OpenSSH.pm:645]: This appears to be a false positive. The substitution produces the standard POSIX shell form `'a'\''b'`, not a backslash inside a single-quoted string. I verified both the unit test and an actual shell round-trip on this branch:\n\n`perl -Ilib t/rsync-quote.t` passes.\n\n`perl -Ilib -MNet::OpenSSH -e 'print Net::OpenSSH->_rsync_quote(q(a'"'"'b)), qq(\n)'` prints `'a'\''b'`.\n\nPassing that quoted string through `sh -c 'printf %s ...'` produces `a'b`.
